### PR TITLE
Modify irfmap stacking

### PR DESCRIPTION
**Pull request by registerrier**
This PR changes the behavior of `PSFMap.stack(other)` and `EDispMap.stack(other)`. 

The two geometries have to be identical in the current version. With this change, the `other` IRF `Map` is reprojected on the target geometry. This is necessary to allow `IRFMapMaker` to use cutouts  in the similar manner as `MapMaker`. See discussion in PR #2056 .

### CODE REVIEW

File: gammapy/cube/psf_map.py [hunk 1]
```diff
@@ -315,17 +315,17 @@ def containment_radius_map(self, energy, fraction=0.68):
 
         return m
 
-    def stack(self, other):
+    def stack(self, other, in_place=True):
```
Comment (adonath) [inline gammapy/cube/psf_map.py]: I think the `in_place` option is not used at the moment. Another suggestion: maybe change to a `copy` option instead? So Always return the stacked map, but in one case a copy is made, whereas in the other `self` is returned. I think this corresponds to the API we have in other places as well e.g. `Map.cutout()`.

Comment (registerrier) [reply gammapy/cube/psf_map.py]: Done.

**Comment by registerrier** [comment]
I have changed to the behavior you suggested for the `copy` parameter for the `stack` method.

I have removed the properties `quantity` `data` and `geom` from the `EDispMap` and the `PSFMap` to avoid the multiple access to `PSFMap.psf_map.quantity` and `PSFMap.quantity`. Only the former is  possible now.

I also updated the docstring example which did not include the calculation of the exposure map.

**Comment by adonath** [comment]
Thanks @registerrier, looks good to me now!

**Pull request merged**